packages/engine/src/input/systems/ClientInputSystem.ts

```diff
@@ -422,8 +422,8 @@ const bboxHitTarget = new Vector3()
 
 const quat = new Quaternion()
 
-const execute = () => {
-  if (!isClient) return null
+const execute = (_isClient = isClient) => {
```

Comment:
we should use `nock` to mock this, rather than modifying runtime code

Reply:
this library https://www.npmjs.com/package/nock or is there another?

Reply:
I think we do need to change runtime code somewhere to consider a Mocha testing enviornment as it is very similar to if we were just running this code on a server. We should consider testing systems in a server, editor, client, and mobile enviornment. Might need to use babel or something to build a testing version where our device booleans are modified with a replace/

Reply:
I really don't think there's a way to do this without dependency injection or changing isClient's implementation to take in account that the environment is running tests. The exported object that contains isClient is just a view of what's exported, there's no way to change it in any meaningful way that would be reflected when the execute function is called, even with a mocking library.

Reply:
Ok. What we should do here is remove the `isClient` dependency of the `ClientInputSystem` execute call. The server should be able to run all the logic here, escaping if data is not available (eg `Engine.instance.cameraEntity`). Same with the reactor, it should add all the hooks to document/window/canvas if it exists. This way we can easily mock all the funcitonality in our tests, without worrying about these `isClient` exports